### ISSUE TITLE
Breaking: False positive from CA1849 in .NET 9 RC1

### ISSUE THREAD

**Issue by bkoelman** [dotnet/roslyn-analyzers#7420]
### Analyzer

**Diagnostic ID**: [CA1849](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1849): `Call async methods when in an async method`

### Analyzer source

**SDK**: Built-in CA analyzers in .NET 9 RC1 SDK
**Version**: 9.0.100-rc.1.24452.12

### Describe the bug

After updating our codebase to .NET 9 RC1, a false positive of CA1849 appears. This doesn't show when building against .NET 8.
The problem is that a warning is produced to use an async method from a lambda that isn't async. So there's no way to fix the reported warning, which makes it a false positive.

### Steps To Reproduce

1. Create a new worker service or web app (so you'll have a host builder in `Program.cs`)
2. Install the (preview version of) the packages `Microsoft.EntityFrameworkCore.Relational` and `Pomelo.EntityFrameworkCore.MySql`.
3. Add the next lines to `Program.cs`:
    ```c#
    string connectionString = "";
    builder.Services.AddMySql<AppDbContext>(connectionString, ServerVersion.AutoDetect(connectionString));

    public sealed class AppDbContext(DbContextOptions<AppDbContext> options) : DbContext(options);
    ```
4. While there is an async version of `ServerVersion.AutoDetect` available, and `Program.Main` is async, the lambda parameter from `AddMySql` is _not_, so this can't be converted to an async call. And no `AddMySqlAsync` method exists.

![image](https://github.com/user-attachments/assets/f9d45486-d316-4490-9309-e32f18566478)

Here's the signature of the `AddMySql` method:
```c#
public static IServiceCollection AddMySql<TContext>(
    this IServiceCollection serviceCollection,
    string connectionString,
    ServerVersion serverVersion,
    Action<MySqlDbContextOptionsBuilder> mySqlOptionsAction = null,
    Action<DbContextOptionsBuilder> optionsAction = null)
    where TContext : DbContext
{
}
```

### Expected behavior
No warning.

### Actual behavior
Warning that can't be fixed.

**Comment by bkoelman** [comment]
My assumption that `AddMySql` takes a lambda for `ServerVersion` was incorrect. It's a regular parameter. So it _can_ be rewritten as:
```c#
ServerVersion serverVersion = await ServerVersion.AutoDetectAsync(connectionString);
builder.Services.AddMySql<AppDbContext>(connectionString, serverVersion);
```

Closing, because it turns out the analyzer is working properly.
